mithril-stm/src/bls_multi_signature/verification_key.rs

```diff
@@ -164,9 +164,9 @@ impl BlsVerificationKeyProofOfPossession {
     /// manually.
     // If we are really looking for performance improvements, we can combine the
     // two final exponentiations (for verifying k1 and k2) into a single one.
-    #[deprecated(since = "0.5.0", note = "Use `verify_proof_of_possesion` instead")]
+    #[deprecated(since = "0.5.0", note = "Use `verify_proof_of_possession` instead")]
```

Comment:
This is probably not a good practice to update a deprecation with a breaking change after the version has been released, and regularly we should keep the `verify_proof_of_possesion` function with the typo and deprecate it. 

On the other hand, the `verify_proof_of_possession` is not public any more (when `check` was) so we can simply update the deprecation note and explain that this function is not available any more.

Reply:
It would have been needed only in the case where the `verify_proof_of_possession` is public. Now it is `pub(crate)` so it is not needed. You can just update the deprecation notice :+1: